thread/thread.cpp

```diff
@@ -896,17 +932,32 @@ R"(
         }
         _photon_switch_context_defer_die(
             arg, func, sw.to->stack.pointer_ref());
+        __builtin_unreachable();
     }
-    static __attribute__((used, noreturn))
+
+    extern "C" __attribute__((noreturn))
+    void _photon_thread_die(thread* th) asm("_photon_thread_die");
     void _photon_thread_die(thread* th) {
         assert(th == CURRENT);
         th->die();
+        __builtin_unreachable();
     }
 
-    extern "C" void _photon_thread_stub() asm ("_photon_thread_stub");
-
-    thread* thread_create(thread_entry start, void* arg,
-                uint64_t stack_size, uint16_t reserved_space) {
+    // the function is written in assembly to make use of the fact
+    // that CURRENT is passed in via rbp/x29 as the first argument,
+    // eliminating a TLS access.
+    extern "C" __attribute__((noreturn))
+    void _photon_thread_stub() asm("_photon_thread_stub");
+
+    thread* thread_create(thread_entry start, void* arg, uint64_t stack_size,
+                uint32_t reserved_space, uint64_t flags) {
+        if (!stack_size)
+            stack_size = DEFAULT_STACK_SIZE;
+        if (unlikely((uint64_t)reserved_space > stack_size / 2))
+            LOG_ERROR_RETURN(EINVAL, nullptr, VALUE(reserved_space), "must be <= half of ", VALUE(stack_size));
+        uint64_t FLAGS = THREAD_JOINABLE | THREAD_ENABLE_WORK_STEALING;
```

Comment:
`FLAGS` looks like just a simple const value, should be defined as const or constexpr.